sample_consensus/include/pcl/sample_consensus/impl/sac_model_circle.hpp

```diff
@@ -175,8 +195,65 @@ pcl::SampleConsensusModelCircle2D<PointT>::countWithinDistance (
     return (0);
   int nr_p = 0;
 
+  std::size_t i = 0;
+#if defined (__AVX__) && defined (__AVX2__)
+  const __m256 a_vec = _mm256_set1_ps (model_coefficients[0]);
+  const __m256 b_vec = _mm256_set1_ps (model_coefficients[1]);
+  // To avoid sqrt computation: consider one larger circle (radius + threshold) and one smaller circle (radius - threshold). Valid if point is in larger circle, but not in smaller circle.
+  const __m256 sqr_inner_circle = _mm256_set1_ps ((model_coefficients[2] <= threshold ? 0.0 : (model_coefficients[2]-threshold)*(model_coefficients[2]-threshold)));
+  const __m256 sqr_outer_circle = _mm256_set1_ps ((model_coefficients[2]+threshold)*(model_coefficients[2]+threshold));
+  __m256i res = _mm256_set1_epi32(0); // This corresponds to nr_p: 8 32bit integers that, summed together, hold the number of inliers
+  for (; (i + 8) <= indices_->size (); i += 8)
+  {
+    const __m256 sqr_dist = sqr_dist8 (i, a_vec, b_vec);
+    const __m256 mask = _mm256_and_ps (_mm256_cmp_ps (sqr_inner_circle, sqr_dist, _CMP_LT_OQ), _mm256_cmp_ps (sqr_dist, sqr_outer_circle, _CMP_LT_OQ)); // The mask contains 1 bits if the corresponding points are inliers, else 0 bits
+    res = _mm256_add_epi32 (res, _mm256_and_si256 (_mm256_set1_epi32 (1), _mm256_castps_si256 (mask))); // The latter part creates a vector with ones (as 32bit integers) where the points are inliers
+    //const int res = _mm256_movemask_ps (mask);
```

Comment:
If comments are required (for explanation), please insert:
```
// equivalent to the following:
```

Else remove unneeded comments